pkg/setup/setup.go

```diff
@@ -293,38 +293,89 @@ func (s *Setup) installDefaultPlugins(httpClient *httpclient.Client) error {
 		return errors.New("DC/OS version of the cluster < 1.10, installation of the plugins aborted")
 	}
 
+	var wg sync.WaitGroup
+	pbar := mpb.New(mpb.WithOutput(s.errout), mpb.WithWaitGroup(&wg))
+	wg.Add(2)
+
 	// Install dcos-enterprise-cli.
-	enterpriseInstallErr := make(chan error)
 	go func() {
 		// Install dcos-enterprise-cli if the DC/OS variant metadata is "enterprise".
 		if version.DCOSVariant == "enterprise" {
-			enterpriseInstallErr <- s.installPlugin("dcos-enterprise-cli", httpClient)
-		}
-		if version.DCOSVariant == "" {
+			if err := s.installPlugin("dcos-enterprise-cli", httpClient, version, pbar); err != nil {
+				s.logger.Debug(err)
+			}
+		} else if version.DCOSVariant == "" {
 			// We add this message if the DC/OS variant is "" (DC/OS < 1.12)
 			// or if there was an error while installing the EE plugin.
 			s.logger.Error("Please run “dcos package install dcos-enterprise-cli” if you use a DC/OS Enterprise cluster")
 		}
-		close(enterpriseInstallErr)
+		wg.Done()
 	}()
 
 	// Install dcos-core-cli.
-	errCore := s.installPlugin("dcos-core-cli", httpClient)
-
-	// The installation of the core and EE plugins happen in parallel.
-	// We wait for the installation of the enterprise plugin before returning.
-	errEnterprise := <-enterpriseInstallErr
-
+	errCore := s.installPlugin("dcos-core-cli", httpClient, version, pbar)
+	wg.Done()
+	pbar.Wait()
 	if errCore != nil {
-		return errCore
+		errCore = corecli.InstallPlugin(s.fs, s.pluginManager)
```

Comment:
Why are we setting `errCore` again if it's not `nil`?

Reply:
I've added a comment, it is about extracting the dcos-core-cli when the installation through S3 / Cosmos failed.